bin/add_celltypes_to_sce.R

```diff
@@ -250,6 +267,47 @@ if (!is.null(opt$cellassign_predictions)) {
     }
     metadata(sce)$cellassign_reference_organs <- cellassign_organs
   }
+} else {
+  has_cellassign <- FALSE # use to decide if we should assign consensus labels
+}
+
+# assign consensus cell type labels
+if (has_singler & has_cellassign) {
+  # now make sure that reference file exists
+  stopifnot(
+    "Consensus cell type reference file does not exist" = file.exists(opt$consensus_celltype_ref)
+  )
+
+  # read in consensus table
+  consensus_ref_df <- readr::read_tsv(opt$consensus_celltype_ref) |>
+    # select unique combinations of consensus refs based on ontology columns
+    dplyr::select(blueprint_ontology, panglao_ontology, consensus_ontology, consensus_annotation) |>
+    unique()
+
+  # create df with consensus assignments
+  celltype_df <- colData(sce) |>
+    as.data.frame() |>
+    dplyr::select(
+      barcodes,
+      contains("celltype") # get both singler and cellassign with ontology
+    ) |>
+    # then add consensus labels
+    dplyr::left_join(
+      consensus_ref_df,
+      by = c(
+        "singler_celltype_ontology" = "blueprint_ontology",
+        "cellassign_celltype_ontology" = "panglao_ontology"
+      ),
+      relationship = "many-to-many" # account for multiple of the same cell type
+    ) |>
+    # use unknown for NA annotation but keep ontology ID as NA
+    # if the sample type is cell line, keep as NA
+    dplyr::mutate(consensus_annotation = dplyr::if_else(is.na(consensus_annotation), "Unknown", consensus_annotation))
```

Comment:
I like `tidyr::replace_na()` for this to be more explicit about what is being done, and much more concise!
I'm confused by the comment about cell lines though, as it doesn't seem that is implemented here?
```suggestion
    tidyr::replace_na(list(consensus_annotation = "Unknown"))
```